common/src/hash.rs

```diff
@@ -86,6 +86,11 @@ pub fn hash_pointer(value: usize) -> PyHash {
     hash as _
 }
 
+#[inline]
+pub fn hash_integer<T: num_traits::PrimInt>(data: T) -> PyHash {
+    fix_sentinel(mod_int(data.to_i64().unwrap()))
```

Comment:
To propagate python exceptions, we don't use `unwrap()` for failable functions.

e.g. what happens if the input is `usize::MAX`?

Reply:
I am not sure what kind of exception (if any) that CPython would throw if I pass in usize::MAX. Will try to look into it.